src/gss_names.c

```diff
@@ -626,9 +708,104 @@ uint32_t gssntlm_inquire_name(uint32_t *minor_status,
                               gss_OID *MN_mech,
                               gss_buffer_set_t *attrs)
 {
-    return GSS_S_UNAVAILABLE;
+    uint32_t retmin;
+    uint32_t retmaj;
+    uint32_t tmpmin;
+    gss_buffer_desc buf;
+    const struct gssntlm_name *in = (const struct gssntlm_name *)name;
+
+    if (name == GSS_C_NO_NAME) {
+        return GSSERRS(GSS_S_BAD_NAME, GSS_S_CALL_INACCESSIBLE_READ);
+    }
+    if (!attrs) {
+        return GSSERRS(ERR_NOARG, GSS_S_CALL_INACCESSIBLE_WRITE);
+    }
+    *attrs = GSS_C_NO_BUFFER_SET;
+
+    if (in->attrs == NULL) {
+        return GSSERRS(0, GSS_S_COMPLETE);
+    }
+
+    for (size_t i = 0; in->attrs[i].attr_name != NULL; i++) {
+        struct gssntlm_name_attribute *attr = &in->attrs[i];
+        size_t attr_name_len = strlen(attr->attr_name);
+
+        gss_buffer_t attr_value = &attr->attr_value;
+        /* +1 for '=' separator and +1 for EOL */
+        buf.length = attr_value->length + attr_name_len + 2;
+        buf.value = malloc(buf.length);
+        if (buf.value == NULL) {
+            retmin = ENOMEM;
+            goto done;
+        }
+
+        /* Construct 'attr_name=<attr_value>\0' string */
+        memcpy(buf.value, attr->attr_name, attr_name_len);
```

Comment:
these casts are heavy to read, can you please instead use a `char *attr_string` (and length) temporary value and use that in this function, assign buf.value = attr_string (and value) at the end only if successful.

Reply:
ok, will do with 'attr_string and length temporary value' but haven't got what is 'only if successful'...
We can't have failures in between of malloc and gss_add_buffer_set_member, and to call the gss_add_buffer_set_member we already need to assign buf.value = attr_string.

Reply:
It had to do with a comment from @frozencemetery that I think applied to this loop, just saying to move error/success handling all in one place and use retmaj to indicate it and goto to the handling part.

Reply:
I have changed this code, pls see the last commit. Is it good enough now?